postgresql/manifests/postgres.yaml

```
@@ -0,0 +1,129 @@
+apiVersion: v1
+kind: Service
+metadata:
+  name: postgres
+  annotations:
+    prometheus.io/scrape: "true"
+    prometheus.io/path: /metrics
+    prometheus.io/port: "9187"
+    uw.health.aggregator.enable: "false"
+spec:
+  ports:
+    - name: db
+      port: 5432
+      protocol: TCP
+    - name: exporter
+      protocol: TCP
+      port: 9187
+  selector:
+    app: postgres
+---
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: postgres
+  labels:
+    app: postgres
+spec:
+  serviceName: postgres
+  replicas: 1
+  selector:
+    matchLabels:
+      app: postgres
+  template:
+    metadata:
+      annotations:
+        prometheus.io/scrape: "true"
+        prometheus.io/path:   /__/metrics
+        prometheus.io/port:   "8081"
+      labels:
+        app: postgres
+      name: postgres
+    spec:
+      containers:
+      - name: metrics
+        env:
+        - name: DATA_SOURCE_NAME
+          valueFrom:
+            secretKeyRef:
+              name: postgres-secrets
+              key: url
+        resources:
+          limits:
+            memory: 64Mi
+            cpu: 100m
+        ports:
+          - containerPort: 9187
+            protocol: TCP
+        imagePullPolicy: IfNotPresent
+        image: quay.io/prometheuscommunity/postgres-exporter:0.15.0
+      - name: postgres
+        image: postgres:16-alpine
+        command:
+          - docker-entrypoint.sh
+          - -c
+          - work_mem=512MB
+          - -c
+          - max_parallel_workers_per_gather=0
+          - -c
+          - max_wal_size=2GB
```

This is a little weird.

The `CMD` value for this container is `["postgres"]`, which starts the database.  The `ENTRYPOINT` value is `["docker-entrypoint.sh"]`, so the above reads as though we'd be running the entrypoint once and then again with additional arguments.  I'm also not sure where the database would actually be started, since this would override the default `CMD` value, right?

Normally, the entrypoint is used to setup/prepare the container to run the command, so it's not clear to me what this actually does.  My guess is what you really want is to override the command and pass these values to `postgres`:

```suggestion
        command:
          - postgres
          - -c
          - work_mem=512MB
          - -c
          - max_parallel_workers_per_gather=0
          - -c
          - max_wal_size=2GB
```

---

I have removed the command overwrite for now. I dont think its worth overwrite in the general case.